### PR TITLE
[state]: infinite recursion bug fix

### PR DESCRIPTION
**Pull request by NVFedorov**
Fix for #280

### CODE REVIEW

File: components/state/src/index.js [hunk 1]
```diff
@@ -68,13 +68,14 @@ AFRAME.registerSystem('state', {
...
   dispatch: (function () {
-    const toUpdate = [];
 
     return function (actionName, payload) {
       var dirtyArrays;
       var i;
       var key;
       var subscription;
+      
+      const toUpdate = [];
```
Comment (ngokevin) [inline components/state/src/index.js:78]: possible to keep this where it was? it was to prevent array allocation on each action

**Comment by NVFedorov** [comment]
good point, fixed

File: components/state/src/index.js (hunk 1 above)
Comment (NVFedorov) [reply components/state/src/index.js]: good point, fixed

File: components/state/src/index.js [hunk 2]
```diff
@@ -96,7 +98,7 @@ AFRAME.registerSystem('state', {
       }
 
       // Notify subscriptions / binders.
-      toUpdate.length = 0;
```
Comment (ngokevin) [inline components/state/src/index.js:99]: maybe this style of array clearing is still okay to use? might be even faster to have less array modification

Comment (NVFedorov) [reply components/state/src/index.js:99]: When we call ```subscriber.onStateUpdate``` and it updates the state inside this method, we will again get inside this method and modify ```toUpdate``` array. Thus we don't want to completely clean up the array, but remove only the subscriber we're notifying right now. Then we need to do something like ```toUpdate.length=toUpdate.length-1```. But ```pop``` does the exact same thing under the hood, as far as I'm concerned.

**Comment by ngokevin** [comment]
sorry for the delay, thanks!

**Comment by NVFedorov** [comment]
Thank you!

**Pull request merged**